pytorch_lightning/trainer/trainer.py

```diff
@@ -1165,14 +1174,17 @@ def _run(
         self._log_hyperparams()
 
         if self.strategy.restore_checkpoint_after_setup:
+            log.detail(f"{self.__class__.__name__}: restoring module and callbacks from checkpoint path: {ckpt_path}")
             self._restore_modules_and_callbacks(ckpt_path)
 
         # restore optimizers, etc.
+        log.detail(f"{self.__class__.__name__}: restoring training state")
         self.checkpoint_connector.restore_training_state()
```

Comment:
I disagree with most changes in this PR. Instead of moving the log calls into the actual methods, this is now littering the trainer with more code that is irrelevant to the reader. Our goal is to make the Trainer more readable and easier to trace steps for contributors and researchers who are curious. This PR is going in the wrong direction IMO.
Please reconsider this, I feel very strongly about this. 

@PytorchLightning/core-contributors

Reply:
cc @PyTorchLightning/core-lightning

Reply:
The choice of logging in the caller vs the callee is as old as logging has existed and it is a tradeoff between flexibility vs convenience.

I don't think these log calls make the Trainer unreadable, as they are just glorified prints. You could see their value as a replacement for comments.

> Please reconsider this, I feel very strongly about this.

However, I don't feel very strongly about it, so if you do, I'm okay with changing these.